lib/inets/src/http_client/httpc_handler.erl

```diff
@@ -1043,15 +1043,29 @@ handle_response(#state{status = new} = State) ->
     ?hcrd("handle response - status = new", []),
     handle_response(try_to_enable_pipeline_or_keep_alive(State));
 
-handle_response(#state{request      = Request,
```

Comment:
I would prefer to avoid using deeply nested cases as it reduces the readability of the functions. In this case it can be replaced by a function that updates state only if there is a `Connection: close` header.

```erlang
handle_response(#state{request      = Request,
		       status       = Status,
		       session      = Session, 
		       status_line  = StatusLine,
		       headers      = Headers, 
		       body         = Body,
		       options      = Options,
		       profile_name = ProfileName} = State0)
  when Status =/= new ->
    State = handle_server_closing(State0),
...
handle_server_closing(State = #state{headers = undefined}) -> State;
handle_server_closing(State = #state{headers = Headers}) ->
    case httpc_response:is_server_closing(Headers) of
        true -> State#state{status = closed};
        false -> State
    end.
```

Reply:
Proposed in new commit. BTW close, not closed.

I plan to run tests locally now.